src/global.ts

```diff
@@ -6,11 +6,10 @@ import type {
 import type { EventTimer } from '.';
 
 declare global {
-	interface Navigator {
-		connection?: {
-			downlink: number;
-			effectiveType: string;
-		};
+	interface NetworkInformation extends EventTarget {
+		readonly type: ConnectionType;
+		readonly downlink?: number;
+		readonly effectiveType?: string;
```

Comment:
Due to a change in `lib.dom.d.ts` from https://github.com/microsoft/TypeScript/pull/45324